backend/services/implementations/classService.ts

```diff
@@ -43,12 +43,14 @@ class ClassService implements IClassService {
         throw new Error("Teacher ID was not found");
       }
 
-      // get the test session details for each test session
-      classObj.testSessions.forEach(async (classTestSession) => {
-        testSessions.push(
-          await this.testSessionService.getTestSessionById(classTestSession),
-        );
-      });
+      await Promise.all(
+        classObj.testSessions.map(async (classTestSession) => {
```

Comment:
.map creates an array of promises, and then Promise.all waits for all the promises to be resolved so that testSessions can be properly initialized before proceeding